### PR TITLE
Implement os.waitpid on Windows

### PR DESCRIPTION
**Pull request by BasixKOR**
This PR implements os.waitpid using MSVCRT's [`_cwait`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/cwait) function.

### CODE REVIEW

**Comment by BasixKOR** [comment]
This function has CPython test suite, although it's not possible to run it on Windows yet.

File: vm/src/stdlib/os.rs [hunk 1]
```diff
@@ -2422,6 +2461,12 @@ fn extend_module_platform_specific(vm: &VirtualMachine, module: &PyObjectRef) {
         "_COPYFILE_DATA" => ctx.new_int(COPYFILE_DATA),
         "_fcopyfile" => ctx.new_function(os_fcopyfile),
     });
+
+    #[cfg(all(windows, target_env = "msvc"))]
+    extend_module!(vm, module, {
+        "wait" => ctx.new_function(os_wait),
+        "waitpid" => ctx.new_function(os_waitpid),
+    })
```
Comment (youknowone) [inline vm/src/stdlib/os.rs:2469]: This is placed in unix `extend_module_platform_specific` for now. Moving this code a few lines below will be perfect.

File: vm/src/stdlib/os.rs [hunk 2]
```diff
@@ -1996,6 +1995,46 @@ fn os_wait(vm: &VirtualMachine) -> PyResult<(libc::pid_t, i32)> {
...
+                "ECHILD: No spawned processes".to_owned(),
+            )),
+            EINVAL => Err(vm.new_exception_msg(
+                vm.ctx.exceptions.os_error.clone(),
+                "EINVAL: Invalid argument".to_owned(),
+            )),
+            _ => unreachable!(),
+        }
+    } else {
+        Ok((pid, status))
```
Comment (BasixKOR) [inline vm/src/stdlib/os.rs:2029]: Oops, this should be lshifted by a byte.

File: vm/src/stdlib/os.rs [hunk 3]
```diff
@@ -2290,6 +2291,46 @@ mod nt {
...
+            unsafe { _get_errno(&mut errno) };
+            match errno {
+                ECHILD => Err(vm.new_exception_msg(
+                    vm.ctx.exceptions.os_error.clone(),
+                    "ECHILD: No spawned processes".to_owned(),
+                )),
+                EINVAL => Err(vm.new_exception_msg(
+                    vm.ctx.exceptions.os_error.clone(),
+                    "EINVAL: Invalid argument".to_owned(),
+                )),
```
Comment (youknowone) [inline vm/src/stdlib/os.rs:2320]: Following unix `convert_nix_error`, here could be `convert_nt_error`

Comment (BasixKOR) [reply vm/src/stdlib/os.rs:2320]: I'd love to use that kind of error handler, but it seems like in `_cwait` Microsoft uses some kind of POSIX-like errno which means it's neither WinAPI error nor POSIX nix error.

**Pull request merged**